docs/v1-proposed-api.md

```diff
@@ -0,0 +1,145 @@
+# Proposed Changes for Hive V1 API
+
+## ClusterDeployment
+
+ 1. Require an InstallConfig ConfigMap, referenced in a new ClusterDeployment.Spec field, to pass directly to installer.
+   * GOAL: Allow quicker exposure to new Installer features without needing to mirror in Hive's API.
+   * GOAL: Remaining ClusterDeployment.Spec should be things that can be modified post-install.
+   * Should Hive parse the InstallConfig to obtain information it needs, eliminating the need for the caller to duplicate information in ClusterDeployment and InstallConfig?
+     * Or should Hive be explicitly given all the information it needs to adopt and maintain a cluster in it's ClusterDeployment, requiring the caller to duplicate a fair portion of InstallConfig in each.
+     * Should we use an InstallConfig if given one, otherwise generate?
+   * Add Spec.InstallConfigMap reference.
+   * Likely cannot embed pull secret and ssh key into the InstallConfig in a ConfigMap.
+     * Should we require the InstallConfig in a Secret instead?
+     * Should we template in the secret values on the fly?
+   * It does not appear that we can remove much of our API. Only Spec.Networking can go. We still need the platform for region, the pull secret + ssh key, the machine sets for on-going management and syncing. There is actually not that much config exposed otherwise.
```

Comment:
Should we get the machinesets from the installer manifests so that they are the completed machinesets after the installer has applied defaults and done its other manipulations? This may fall in line well with moving the machineset syncing to syncsets.